src/imageLoader/convertColorSpace.js

```diff
@@ -7,38 +7,38 @@ import {
   convertPALETTECOLOR,
 } from './colorSpaceConverters/index.js';
 
-function convertRGB(imageFrame, rgbaBuffer) {
+function convertRGB(imageFrame, colorBuffer, useRGBA) {
   if (imageFrame.planarConfiguration === 0) {
-    convertRGBColorByPixel(imageFrame.pixelData, rgbaBuffer);
+    convertRGBColorByPixel(imageFrame.pixelData, colorBuffer, useRGBA);
   } else {
-    convertRGBColorByPlane(imageFrame.pixelData, rgbaBuffer);
+    convertRGBColorByPlane(imageFrame.pixelData, colorBuffer, useRGBA);
   }
 }
 
-function convertYBRFull(imageFrame, rgbaBuffer) {
+function convertYBRFull(imageFrame, colorBuffer, useRGBA) {
   if (imageFrame.planarConfiguration === 0) {
-    convertYBRFullByPixel(imageFrame.pixelData, rgbaBuffer);
+    convertYBRFullByPixel(imageFrame.pixelData, colorBuffer, useRGBA);
   } else {
-    convertYBRFullByPlane(imageFrame.pixelData, rgbaBuffer);
+    convertYBRFullByPlane(imageFrame.pixelData, colorBuffer, useRGBA);
   }
 }
 
-export default function convertColorSpace(imageFrame, imageData) {
-  const rgbaBuffer = imageData.data;
-  // convert based on the photometric interpretation
+export default function convertColorSpace(imageFrame, imageData, useRGBA) {
+  const colorBuffer = useRGBA ? imageData.data : imageData;
```

Comment:
this is pretty weird, can't we just make the input of this function a typed array?